src/compiler/utilities.ts

```diff
@@ -758,6 +758,10 @@ namespace ts {
         return node.kind !== SyntaxKind.JsxText ? getLeadingCommentRanges(sourceFileOfNode.text, node.pos) : undefined;
     }
 
+    export function getTrailingCommentRangesOfNode(node: Node, sourceFileOfNode: SourceFile) {

```

Comment:
This is misleadingly named - the trailing comment ranges of the node would be the trailing comment ranges of the `end` of the node - this is the trailing comment ranges working backwards from the beginning of the node - which are actually the trailing comment ranges of the end of _the prior token_. Also, consider the quite crazily formatted:
```ts
class A {
   constructor(
        public foo: string, /* @internal */
        // This next one
        public bar: number,
  )
}
```
I don't _think_ that should count as an internal annotation for the second parameter declaration, due to the intervening single-line comment. (On a related note, I'd love to see a few more tests for some odd internal comment locations around parameter properties with this change).